gap/attributes/dual.gd

```diff
@@ -0,0 +1,25 @@
+#############################################################################
+##
+#W  dual.gd
+#Y  Copyright (C) 2017                                      James D. Mitchell
+##                                                          Finn Smith
+##
+##  Licensing information can be found in the README file of this package.
+##
+#############################################################################
+##
+
+DeclareCategory("IsDualSemigroupElement", IsAssociativeElement);
+DeclareCategoryCollections("IsDualSemigroupElement");
+DeclareAttribute("DualSemigroup", IsSemigroup);
+DeclareSynonym("IsDualSemigroup", IsSemigroup and
+                                  IsDualSemigroupElementCollection);
+DeclareAttribute("TypeDualSemigroupElements", IsDualSemigroup);
```

Comment:
The type and the family should probably be internal parts of the representation of `IsDualSemigroup` rather than visible attributes as here.